disnake/guild_scheduled_event.py

```diff
@@ -0,0 +1,372 @@
+
+if TYPE_CHECKING:
+    from .abc import GuildChannel
+    from .guild import Guild
+    from .state import ConnectionState
+
+
+__all__ = ("GuildScheduledEventMetadata", "GuildScheduledEvent")
+
+
+class GuildScheduledEventMetadata:
+    """
+    Represents guild event entity metadata.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    location: Optional[:class:`str`]
+        Location of the event
+    """
+
+    __slots__ = ("location",)
+
+    def __init__(self, *, location: str = None):
+        self.location: Optional[str] = location
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"location": self.location}
+
+    @classmethod
+    def from_dict(cls, data):
+        return GuildScheduledEventMetadata(location=data.get("location"))
+
+
+class GuildScheduledEvent(Hashable):
+    """
+    Represents guild scheduled events.
+
+    .. versionadded:: 2.3
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two scheduled events are equal.
+
+        .. describe:: x != y
+
+            Checks if two scheduled events are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the scheduled event's hash.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The id of the scheduled event.
+    guild_id: :class:`int`
+        The guild id which the scheduled event belongs to.
+    channel_id: Optional[:class:`int`]
+        The channel id in which the scheduled event will be hosted.
+        This field is ``None`` if :attr:`entity_type` is :class:`GuildScheduledEventEntityType.external`
+    creator_id: Optional[:class:`int`]
+        The id of the user that created the scheduled event.
+        This field is ``None`` for events created before October 25th, 2021.
+    name: :class:`str`
+        The name of the scheduled event (1-100 characters).
+    description: :class:`str`
+        The description of the scheduled event (1-1000 characters).
+    scheduled_start_time: :class:`datetime`
+        The time the event will start.
+    scheduled_end_time: Optional[:class:`datetime`]
+        The time the event will end, or ``None`` if the event does not have a scheduled time to end.
+    privacy_level: :class:`StagePrivacyLevel`
+        The privacy level of the scheduled event.
+    status: :class:`GuildScheduledEventStatus`
+        The status of the scheduled event.
+    entity_type: :class:`GuildScheduledEventEntityType`
+        The type of the scheduled event.
+    entity_id: Optional[:class:`int`]
+        The id of an entity associated with a guild scheduled event.
+    entity_metadata: :class:`GuildScheduledEventMetadata`
+        Additional metadata for the guild scheduled event.
+    creator: Optional[:class:`User`]
+        The user that created the scheduled event.
+        This field is ``None`` for events created before October 25th, 2021.
+    user_count: Optional[:class:`int`]
```

Comment:
Can this generally have a value, or is it only non-null if `with_user_count=True` is set? If so, the docstring should mention this dependency